x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/watcher/history/WatchRecord.java

```diff
@@ -356,5 +418,104 @@ void innerToXContent(XContentBuilder builder, Params params) throws IOException
                 }
             }
         }
+
+        @Override
+        public WatchRecord dropLargeFields() throws Exception {
+            return new ExceptionWatchRecord(
+                this.id,
+                triggerEvent,
+                this.state,
+                this.vars,
+                this.input == null ? null : getTruncatedInput(),
+                this.condition,
+                this.metadata,
+                this.watch,
+                this.executionResult == null ? null : getTruncatedWatchExecutionResult(this),
+                this.getNodeId(),
+                this.user,
+                this.exception
+            );
+        }
+    }
+
+    /*
+     * This returns a ExecutableInput whose toXContent() returns no information other than a new TRUNCATED_MESSAGE field. It
+     * drops other information to avoid having a document that is too large to index into Elasticsearch.
+     */
+    private static ExecutableInput<? extends Input, ? extends Input.Result> getTruncatedInput() {
+        return new ExecutableInput<>(new Input() {
+            @Override
+            public String type() {
+                return TRUNCATED_RECORD_KEY;
+            }
+
+            @Override
+            public XContentBuilder toXContent(XContentBuilder builder, Params params) throws IOException {
+                builder.value(TRUNCATED_RECORD_VALUE);
+                // builder.startObject();
+                // builder.field(TRUNCATED_RECORD_KEY, TRUNCATED_RECORD_VALUE);
+                // builder.endObject();
+                return builder;
```

Comment:
I'm assuming a string here isn't going to cause any issues mapping wise?

Reply:
Good question. I'll go double-check that the mapping is dynamic (I haven't run into any problems doing this, so I assume it is). It's actually not just writing the string here. Oddly, it's called from here (https://github.com/elastic/elasticsearch/blob/main/x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/watcher/client/WatchSourceBuilder.java#L169), and it sets the key to type(), and effectively sets the value to toXContent().
Also I'll delete those commented-out lines I apparently left in.